packages/common/src/types.ts

```diff
@@ -353,10 +381,59 @@ export interface RANGESection {
 }
 
 /**
- * A Schedule
- * @param years a list of the years of this object
- * @param yearMap an object containing the year objects of this schedule
- * @param id the id number of this schedule
+ * A clean version of a student's schedule as used in V2 of the App with no
+ * redundunt year information.
+ */
+export interface Schedule2<T> {
+  years: ScheduleYear2<T>[];
+}
+
+export interface ScheduleYear2<T> {
+  year: number;
+  fall: ScheduleTerm2<T>;
+  spring: ScheduleTerm2<T>;
+  summer1: ScheduleTerm2<T>;
+  summer2: ScheduleTerm2<T>;
+  isSummerFull: boolean;
+}
+
+/**
+ * A clean version of the ScheduleTerm used by V2 of the App. A generic id field
+ * is used for book keeping purposes by the drag and drop library, in cases
+ * where we don't care about this id, T can null.
+ */
+export interface ScheduleTerm2<T> {
```

Comment:
A little confused on why we need a generic type

Reply:
good point, we could get away with the id field being string | null. alex's design had generic to allow the type to be more flexible, in case we want the book keeping id field to be more complicate(maybe an object w multiple fields).

i think string | null is a good choice and while it isn't fully clear as to which option is better, both options work and aren't too bad so imma leave it as is